src/index.js

```diff
@@ -14,26 +14,39 @@ class TemplateContentScript extends ContentScript {
   // ////////
   async ensureAuthenticated() {
     this.log('debug', 'Starting ensureAuthenticated')
-    const credentials = await this.getCredentials()
-    if (credentials) {
-      const auth = await this.authWithCredentials(credentials)
-      if (auth) {
+    await this.goto(BASE_URL)
+    await this.waitForElementInWorker('.inside-space')
+    await this.ensureNotAuthenticated()
+    const authenticated = await this.runInWorker('checkAuthenticated')
+    if (!authenticated) {
+      this.log('info', 'Not authenticated')
+      const credentials = await this.getCredentials()
+      if (credentials) {
+        await this.authWithCredentials(credentials)
         return true
       }
-      return false
+      await this.authWithoutCredentials()
     }
-    if (!credentials) {
-      const auth = await this.authWithoutCredentials()
-      if (auth) {
-        return true
-      }
-      return false
+    return true
+  }
+
+  async ensureNotAuthenticated() {
+    this.log('debug', 'Starting ensureNOTAuthenticated')
+    const authenticated = await this.runInWorker('checkAuthenticated')
```

Comment:
Without any goto here, there will be nothing to check

Reply:
Yes there is, on veolia there is no specific login page, the login is made on the main page so as we always load the main page before everything else, we're good.

Reply:
No, ensureNotAuthenticated can be called [before even ensureAuthenticated](https://github.com/cozy/cozy-flagship-app/blob/4844fcf0b981a975c2490882d88f35fafdf0c95d/src/libs/ReactNativeLauncher.js#L239-L253) on first run when credentials have been cleaned. You need to be sure to have a veolia page loaded.